src/infrastructure/YamlGenericTemplateStorage.go

```diff
@@ -0,0 +1,526 @@
+			}
+			return reflect.Indirect(firstReflect).String() < reflect.Indirect(secondReflect).String()
+
+		case reflect.Int:
+			if strings.ToLower(orderDirection) == "desc" || strings.ToLower(orderDirection) == "descending" {
+				return reflect.Indirect(firstReflect).Int() > reflect.Indirect(secondReflect).Int()
+			}
+			return reflect.Indirect(firstReflect).Int() < reflect.Indirect(secondReflect).Int()
+
+		case reflect.Struct:
+			if firstReflect.Type().String() == "time.Time" {
+				fTime := firstReflect.Interface().(time.Time)
+				sTime := secondReflect.Interface().(time.Time)
+				if strings.ToLower(orderDirection) == "desc" || strings.ToLower(orderDirection) == "descending" {
+					return fTime.After(sTime)
+				}
+				return fTime.Before(sTime)
+			}
+			return false
+
+		default:
+			return false
+		}
+	})
+	return nil
+}
+
+//GetByName gets template by name
+//Params:
+//	ctx - context is used only for logging
+//	templateName - name of template
+//Return:
+//	*TemplateType - pointer to template
+//	error - if an error occurs, otherwise nil
+func (y *YamlGenericTemplateStorage[TemplateType]) GetByName(ctx context.Context, templateName string) (*TemplateType, error) {
+	y.log(ctx, logrus.DebugLevel, fmt.Sprintf("GetByName: name = %s", templateName))
+	return y.getTemplateObjFromYaml(templateName)
+}
+
+//GetList gets list of templates with filtering and pagination
+//
+//Params:
+//	ctx - context is used only for logging
+//	orderBy - order by string parameter
+//	orderDirection - ascending or descending order
+//	page - page number
+//	size - page size
+//	queryBuilder - query builder for filtering
+//Return:
+//	*[]TemplateType - pointer to array of templates
+//	error - if an error occurs, otherwise nil
+func (y *YamlGenericTemplateStorage[TemplateType]) GetList(ctx context.Context, orderBy, orderDirection string, page, pageSize int, queryBuilder interfaces.IQueryBuilder) (*[]TemplateType, error) {
+	y.log(ctx, logrus.DebugLevel, fmt.Sprintf("GetList: IN: orderBy=%s, orderDirection=%s, page=%d, size=%d, queryBuilder=%s", orderBy, orderDirection, page, pageSize, queryBuilder))
+	var templatesSlice []TemplateType
+	offset := (page - 1) * pageSize
+	files, err := ioutil.ReadDir(y.TemplatesDirectory)
+	if err != nil {
+		return nil, err
+	}
+	for _, f := range files {
+		template, err := y.getTemplateObjFromYaml(f.Name())
+		if err != nil {
+			return nil, err
+		}
+		templatesSlice = append(templatesSlice, *template)
+	}
+
+	var templates *[]TemplateType
+	if queryBuilder != nil {
+		queryStr, err := queryBuilder.Build()
+		if err != nil {
+			return nil, err
+		}
+		queryArr := queryStr.([]interface{})
+		templates, err = y.handleQuery(templatesSlice, queryArr...)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		templates = &templatesSlice
+	}
+
+	err = y.sortTemplatesSlice(templates, orderBy, orderDirection)
+	if err != nil {
+		return nil, err
```

Comment:
We should add more information about what kind of error happened. For example, add to the beginning of the error: "sorting error: ", because above, we must understand at what action an error occurred.